From 4e54a6ac53b855c1f0d51f9a44bf0af5a68e2f3d Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Thu, 3 Sep 2015 22:16:35 +0200
Subject: [PATCH 26/33] libgcc: epiphany: crti.S: Respect
 __USER_LABEL_PFREFIX__

libgcc:
	* config/epiphany/crti.S (_init) (_fini): Respect
	__USER_LABEL_PREFIX__.

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 libgcc/ChangeLog.epiphany     |  5 +++++
 libgcc/config/epiphany/crti.S | 10 ++++++----
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/libgcc/ChangeLog.epiphany b/libgcc/ChangeLog.epiphany
index 20b77967a37..adea2d3b459 100644
--- a/libgcc/ChangeLog.epiphany
+++ b/libgcc/ChangeLog.epiphany
@@ -1,3 +1,8 @@
+2015-09-03  Ola Jeppsson  <ola@adapteva.com>
+
+	* config/epiphany/crti.S (_init) (_fini): Respect
+	__USER_LABEL_PREFIX__.
+
 2015-09-03  Ola Jeppsson  <ola@adapteva.com>
 	* config/epiphany/crti.S: Add underscore prefix to init and
 	fini functions so that we're still namespace clean.
diff --git a/libgcc/config/epiphany/crti.S b/libgcc/config/epiphany/crti.S
index b9a71bc4a4b..c3a48aca550 100644
--- a/libgcc/config/epiphany/crti.S
+++ b/libgcc/config/epiphany/crti.S
@@ -21,14 +21,16 @@
 # see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 # <http://www.gnu.org/licenses/>.
 
+#include "epiphany-asm.h"
+
 	.section .init
-	.global	_init
+	.global	SYM(_init)
 	.balign 2
-_init:
+SYM(_init):
 	str lr,[sp],-4
 
 	.section .fini
-	.global	_fini
+	.global	SYM(_fini)
 	.balign 2
-_fini:
+SYM(_fini):
 	str lr,[sp],-4
-- 
2.31.1

